File: PST/FUEL Source Code/FUEL/BigRed/TestSuite.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.BigRed.TestSuite
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using Microsoft.Office.Interop.Excel;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Diagnostics;

namespace FUEL.BigRed
{
  public class TestSuite
  {
    public TestSuite()
    {
      this.TemplateRev = 0;
      this.Variables = new TestVariables();
    }

    public int TemplateRev { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public TestVariables Variables { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public string[,] Instructions { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public void LoadNativeSuite(string Path)
    {
      Workbook workbook = ExcelHandler.LoadExcel(Path);
      this.TemplateRev = Conversions.ToInteger(NewLateBinding.LateGet(NewLateBinding.LateIndexGet(workbook.BuiltinDocumentProperties, new object[1]
      {
        (object) "Revision Number"
      }, (string[]) null), (Type) null, "value", new object[0], (string[]) null, (Type[]) null, (bool[]) null));
      object[,] arrVariables = ExcelHandler.ReadSheet(workbook, 1);
      object[,] arrInstructions = ExcelHandler.ReadSheet(workbook, 2);
      ExcelHandler.CloseWB(workbook);
      this.Variables = this.LoadVariables(arrVariables);
      this.Instructions = this.LoadInstructions(arrInstructions);
    }

    private TestVariables LoadVariables(object[,] arrVariables) => new TestVariables()
    {
      AutoDataOnSchedule = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Run AutoData on a schedule?")),
      PassiveAutoData = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Use Passive AutoData?")),
      UploadDataToGradeBook = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Upload data to GradeBook?")),
      TestRunningOutsideOfHPFireWall = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Test running outside of HP Firewall?")),
      CollectDataForEveryPrintJob = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Collect data for every print job?")),
      SendDataViaEmail = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Send data via email?")),
      EmailAddress = this.Variables_Set(arrVariables, "To whom do you want email sent?"),
      MessagePriority = Conversions.ToInteger(this.Variables_Set(arrVariables, "Messaging Priority")),
      ReplacePgNrWithPlotCD = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Replace Page Nr with PLOT_CD?")),
      PrintProtocol = this.Variables_Set(arrVariables, "Select Print Protocols"),
      AutomaticeLIDILMode = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Automatic eLIDIL Mode")),
      PCLFileLocation = this.Variables_Set(arrVariables, "PCL Location"),
      NozzlePatternLocation = this.Variables_Set(arrVariables, "Nozzle Pattern Locations"),
      AddCRLFToHeaders = Conversions.ToBoolean(this.Variables_Set(arrVariables, "Auto add Carriage Return to Header?")),
      FooterData = this.Variables_DataCollections(arrVariables, "Footer Data"),
      GradeBookData = this.Variables_DataCollections(arrVariables, "GradeBook Data")
    };

    private string Variables_Set(object[,] arr, string LookinFor)
    {
      int num1 = Information.UBound((Array) arr);
      int index = 1;
      while (index <= num1)
      {
        if (Operators.ConditionalCompareObjectNotEqual(arr[index, 1], (object) null, false) && Operators.ConditionalCompareObjectEqual(NewLateBinding.LateGet(NewLateBinding.LateGet(arr[index, 1], (Type) null, "trim", new object[0], (string[]) null, (Type[]) null, (bool[]) null), (Type) null, "tolower", new object[0], (string[]) null, (Type[]) null, (bool[]) null), (object) LookinFor.Trim().ToLower(), false))
          return Conversions.ToString(arr[index, 2]);
        checked { ++index; }
      }
      int num2 = (int) Interaction.MsgBox((object) ("Error: Unable to find '" + LookinFor + "' in test variables."), MsgBoxStyle.Critical, (object) "Cant Find Test Variable");
      return Conversions.ToString(0);
    }

    private string[] Variables_DataCollections(object[,] arr, string LookinFor)
    {
      bool flag = false;
      int index1 = 0;
      int num1 = Information.UBound((Array) arr);
      int index2 = 1;
      while (index2 <= num1)
      {
        if (Operators.ConditionalCompareObjectNotEqual(arr[index2, 1], (object) null, false) && Operators.ConditionalCompareObjectEqual(NewLateBinding.LateGet(NewLateBinding.LateGet(arr[index2, 1], (Type) null, "trim", new object[0], (string[]) null, (Type[]) null, (bool[]) null), (Type) null, "tolower", new object[0], (string[]) null, (Type[]) null, (bool[]) null), (object) LookinFor.Trim().ToLower(), false))
        {
          index1 = index2;
          flag = true;
          break;
        }
        checked { ++index2; }
      }
      if (!flag)
        return (string[]) null;
      int num2 = 0;
      int num3 = Information.UBound((Array) arr, 2);
      int index3 = 1;
      while (index3 <= num3)
      {
        if (Operators.ConditionalCompareObjectEqual(arr[index1, index3], (object) null, false))
        {
          num2 = checked (index3 - 1);
          break;
        }
        checked { ++index3; }
      }
      string[] strArray = new string[checked (num2 - 3 + 1)];
      int num4 = num2;
      int index4 = 3;
      while (index4 <= num4)
      {
        strArray[checked (index4 - 3)] = Conversions.ToString(arr[index1, index4]);
        checked { ++index4; }
      }
      return strArray;
    }

    private string[,] LoadInstructions(object[,] arrInstructions)
    {
      int num1 = 0;
      int num2 = Information.UBound((Array) arrInstructions);
      int index1 = 1;
      while (index1 <= num2)
      {
        if (Operators.ConditionalCompareObjectEqual(arrInstructions[index1, 1], (object) null, false))
        {
          num1 = checked (index1 - 1);
          break;
        }
        checked { ++index1; }
      }
      int num3 = 0;
      int num4 = Information.UBound((Array) arrInstructions, 2);
      int index2 = 1;
      while (index2 <= num4)
      {
        if (Operators.ConditionalCompareObjectNotEqual(arrInstructions[1, index2], (object) null, false))
          num3 = index2;
        checked { ++index2; }
      }
      string[,] strArray = new string[checked (num1 - 2 + 1), checked (num3 - 1 + 1)];
      int num5 = num1;
      int index3 = 2;
      while (index3 <= num5)
      {
        int num6 = num3;
        int index4 = 1;
        while (index4 <= num6)
        {
          strArray[checked (index3 - 2), checked (index4 - 1)] = !Operators.ConditionalCompareObjectEqual(arrInstructions[index3, index4], (object) null, false) ? Conversions.ToString(arrInstructions[index3, index4]) : "";
          checked { ++index4; }
        }
        checked { ++index3; }
      }
      return strArray;
    }
  }
}

File: PST/FUEL/src/FUEL/BigRed/TestSuite.vb
Imports FUEL
Imports Microsoft.Office.Interop.Excel
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Diagnostics
Imports System.Runtime.CompilerServices

Namespace FUEL.BigRed
    Public Class TestSuite
        ' Methods
        Public Sub New()
            Me.TemplateRev = 0
            Me.Variables = New TestVariables
        End Sub

        Private Function LoadInstructions(ByVal arrInstructions As Object(0 To .,0 To .)) As String(0 To .,0 To .)
            Dim num2 As Integer = 0
            Dim num7 As Integer = Information.UBound(arrInstructions, 1)
            Dim num3 As Integer = 1
            Do While True
                Dim flag As Boolean
                Dim num11 As Integer = num7
                If (num3 <= num11) Then
                    flag = Operators.ConditionalCompareObjectEqual(arrInstructions(num3, 1), Nothing, False)
                    If Not flag Then
                        num3 += 1
                        Continue Do
                    End If
                    num2 = (num3 - 1)
                End If
                Dim num As Integer = 0
                Dim num8 As Integer = Information.UBound(arrInstructions, 2)
                Dim num4 As Integer = 1
                Do While True
                    num11 = num8
                    If (num4 > num11) Then
                        Dim strArray2(,) As String(0 To .,0 To .) = New String(((num2 - 2) + 1)  - 1, ((num - 1) + 1)  - 1) {}
                        Dim num9 As Integer = num2
                        Dim num5 As Integer = 2
                        Do While (num5 <= num9)
                            Dim num10 As Integer = num
                            Dim num6 As Integer = 1
                            Do While True
                                num11 = num10
                                If (num6 > num11) Then
                                    num5 += 1
                                    Exit Do
                                End If
                                flag = Operators.ConditionalCompareObjectEqual(arrInstructions(num5, num6), Nothing, False)
                                strArray2((num5 - 2), (num6 - 1)) = If(Not flag, Conversions.ToString(arrInstructions(num5, num6)), "")
                                num6 += 1
                            Loop
                        Loop
                        Return strArray2
                    End If
                    If Operators.ConditionalCompareObjectNotEqual(arrInstructions(1, num4), Nothing, False) Then
                        num = num4
                    End If
                    num4 += 1
                Loop
            Loop
        End Function

        Public Sub LoadNativeSuite(ByVal Path As String)
            Dim wB As Workbook = ExcelHandler.LoadExcel(Path)
            Dim arguments As Object() = New Object() { "Revision Number" }
            Me.TemplateRev = Conversions.ToInteger(NewLateBinding.LateGet(NewLateBinding.LateIndexGet(wB.get_BuiltinDocumentProperties, arguments, Nothing), Nothing, "value", New Object(0  - 1) {}, Nothing, Nothing, Nothing))
            Dim arrVariables(,) As Object(0 To .,0 To .) = ExcelHandler.ReadSheet(wB, 1)
            Dim arrInstructions(,) As Object(0 To .,0 To .) = ExcelHandler.ReadSheet(wB, 2)
            ExcelHandler.CloseWB(wB)
            Me.Variables = Me.LoadVariables(arrVariables)
            Me.Instructions = Me.LoadInstructions(arrInstructions)
        End Sub

        Private Function LoadVariables(ByVal arrVariables As Object(0 To .,0 To .)) As TestVariables
            Return New TestVariables With { _
                .AutoDataOnSchedule = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Run AutoData on a schedule?")), _
                .PassiveAutoData = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Use Passive AutoData?")), _
                .UploadDataToGradeBook = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Upload data to GradeBook?")), _
                .TestRunningOutsideOfHPFireWall = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Test running outside of HP Firewall?")), _
                .CollectDataForEveryPrintJob = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Collect data for every print job?")), _
                .SendDataViaEmail = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Send data via email?")), _
                .EmailAddress = Me.Variables_Set(arrVariables, "To whom do you want email sent?"), _
                .MessagePriority = Conversions.ToInteger(Me.Variables_Set(arrVariables, "Messaging Priority")), _
                .ReplacePgNrWithPlotCD = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Replace Page Nr with PLOT_CD?")), _
                .PrintProtocol = Me.Variables_Set(arrVariables, "Select Print Protocols"), _
                .AutomaticeLIDILMode = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Automatic eLIDIL Mode")), _
                .PCLFileLocation = Me.Variables_Set(arrVariables, "PCL Location"), _
                .NozzlePatternLocation = Me.Variables_Set(arrVariables, "Nozzle Pattern Locations"), _
                .AddCRLFToHeaders = Conversions.ToBoolean(Me.Variables_Set(arrVariables, "Auto add Carriage Return to Header?")), _
                .FooterData = Me.Variables_DataCollections(arrVariables, "Footer Data"), _
                .GradeBookData = Me.Variables_DataCollections(arrVariables, "GradeBook Data") _
            }
        End Function

        Private Function Variables_DataCollections(ByVal arr As Object(0 To .,0 To .), ByVal LookinFor As String) As String()
            Dim flag As Boolean = False
            Dim num2 As Integer = 0
            Dim num6 As Integer = Information.UBound(arr, 1)
            Dim num3 As Integer = 1
            Do While True
                Dim strArray2 As String()
                Dim num9 As Integer = num6
                If (num3 <= num9) Then
                    Dim flag2 As Boolean = Operators.ConditionalCompareObjectNotEqual(arr(num3, 1), Nothing, False)
                    If (Not flag2 OrElse Not Operators.ConditionalCompareObjectEqual(NewLateBinding.LateGet(NewLateBinding.LateGet(arr(num3, 1), Nothing, "trim", New Object(0  - 1) {}, Nothing, Nothing, Nothing), Nothing, "tolower", New Object(0  - 1) {}, Nothing, Nothing, Nothing), LookinFor.Trim.ToLower, False)) Then
                        num3 += 1
                        Continue Do
                    End If
                    num2 = num3
                    flag = True
                End If
                If Not flag Then
                    strArray2 = Nothing
                Else
                    Dim num As Integer = 0
                    Dim num7 As Integer = Information.UBound(arr, 2)
                    Dim num4 As Integer = 1
                    Do While True
                        num9 = num7
                        If (num4 <= num9) Then
                            If Not Operators.ConditionalCompareObjectEqual(arr(num2, num4), Nothing, False) Then
                                num4 += 1
                                Continue Do
                            End If
                            num = (num4 - 1)
                        End If
                        Dim strArray As String() = New String(((num - 3) + 1)  - 1) {}
                        Dim num8 As Integer = num
                        Dim num5 As Integer = 3
                        Do While True
                            num9 = num8
                            If (num5 > num9) Then
                                strArray2 = strArray
                                Exit Do
                            End If
                            strArray((num5 - 3)) = Conversions.ToString(arr(num2, num5))
                            num5 += 1
                        Loop
                        Exit Do
                    Loop
                End If
                Return strArray2
            Loop
        End Function

        Private Function Variables_Set(ByVal arr As Object(0 To .,0 To .), ByVal LookinFor As String) As String
            Dim num2 As Integer = Information.UBound(arr, 1)
            Dim num As Integer = 1
            Do While True
                Dim str2 As String
                Dim num3 As Integer = num2
                If (num > num3) Then
                    Interaction.MsgBox(("Error: Unable to find '" & LookinFor & "' in test variables."), MsgBoxStyle.Critical, "Cant Find Test Variable")
                    str2 = Conversions.ToString(0)
                Else
                    Dim flag As Boolean = Operators.ConditionalCompareObjectNotEqual(arr(num, 1), Nothing, False)
                    If (Not flag OrElse Not Operators.ConditionalCompareObjectEqual(NewLateBinding.LateGet(NewLateBinding.LateGet(arr(num, 1), Nothing, "trim", New Object(0  - 1) {}, Nothing, Nothing, Nothing), Nothing, "tolower", New Object(0  - 1) {}, Nothing, Nothing, Nothing), LookinFor.Trim.ToLower, False)) Then
                        num += 1
                        Continue Do
                    End If
                    str2 = Conversions.ToString(arr(num, 2))
                End If
                Return str2
            Loop
        End Function


        ' Properties
        Public Property TemplateRev As Integer
            <DebuggerNonUserCode> _
            Get
                Return Me._TemplateRev
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As Integer)
                Me._TemplateRev = AutoPropertyValue
            End Set
        End Property

        Public Property Variables As TestVariables
            <DebuggerNonUserCode> _
            Get
                Return Me._Variables
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As TestVariables)
                Me._Variables = AutoPropertyValue
            End Set
        End Property

        Public Property Instructions As String(0 To .,0 To .)
            <DebuggerNonUserCode> _
            Get
                Return Me._Instructions
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String(0 To .,0 To .))
                Me._Instructions = AutoPropertyValue
            End Set
        End Property


        ' Fields
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private _TemplateRev As Integer
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private _Variables As TestVariables
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private _Instructions As String(0 To .,0 To .)
    End Class
End Namespace

